File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution{
    public:
        vector <int> twoSum(vector<int>&nums, int target){
            if(nums.size() < 2){
                return {nums[0]}; 
            }

            for(int i = 0; i < nums.size(); i++){
                for(int k = i + 1; k < nums.size(); k++){
                    if(nums[i] + nums[k] == target){
                        return {i, k}; 
                    }
                }
            }
            return {}; 
        }
};   

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0, r = nums.length - 1;

        while(l <= r){
            int middle = (l+r) / 2;
            if(nums[middle] > target){
                r = middle - 1;
            }
            else if(nums[middle] < target){
                l = middle + 1;
            }
            else {
                return middle;
            }
        }

        return -1;
    }
}
